internal/host/validator.go

```diff
@@ -1229,18 +1229,179 @@ func (v *validator) printDefaultRoute(c *validationContext, status ValidationSta
 	}
 }
 
-func shouldValidateDnsResolution(c *validationContext) bool {
-	// Skip DNS resolution checks in IPI network mode
-	if !swag.BoolValue(c.cluster.UserManagedNetworking) {
+func ignitionHasFile(ignition *ignition_types.Config, path string) bool {
+	for _, file := range ignition.Storage.Files {
+		if file.Path == path {
+			return true
+		}
+	}
+
+	return false
+}
+
+func ignitionContainsManagedNetworkingFiles(config *ignition_types.Config) bool {
+	if ignitionHasFile(config, "/etc/kubernetes/manifests/coredns.yaml") ||
+		ignitionHasFile(config, "/etc/kubernetes/manifests/keepalived.yaml") {
+		// This is a hack - since we have no official way to know whether a
+		// worker ignition file originated from a cluster with managed
+		// networking or not, we instead rely on the presence of coredns and
+		// keepalived pod manifests to indicate that. We only expect those to
+		// be present in clusters with managed networking. To be a bit more
+		// robust, we consider the presence of any one of them to mean that the
+		// cluster has managed networking. This gives us better forwards
+		// compatibility if either one of them gets renamed / replaced with
+		// other technologies in future OCP versions.
+		//
+		// Another way in which this is hacky is that users could manually
+		// create static pods with the same name as part of their
+		// machine-configs, in which case we would have a false-positive
+		// detection. But that is admittedly very unlikely.
+		//
+		// Hopefully we can negotiate with the relevant OCP teams to have a
+		// more official, stable way to have this detection - like a magic
+		// empty file placed somewhere in the ignition that we can check for
+		// the presence of. Once we have such file, we can slowly deprecate
+		// this detection mechanism and fully move to the new one by inspecting
+		// that file instead.
```

Comment:
I think this longer explanation is probably better in the commit. I think something smaller would be good for a comment if you can come up with a useful smaller description.